### PR TITLE
doc/architecture: security documentation (fixes: #106)

### PR DESCRIPTION
**Pull request by Hjortsberg**
Initial documentation of security measures.

### CODE REVIEW

File: doc/architecture/security.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+TLS
+---
+The connection from the client to the web server is made over TLS with a Amazon signed certificate.
+This is implemented on the Elastic Loadbalancer of Elastic Beanstalk.
+The request then propagates through our VPC (Virtual private cloud) in AWS as plain HTTP. 
+
+CSRF
+----
+The site has implemented protection against CSRF (Cross Site Request Forgery), 
+> this is a manin the middle attack that abuses the browser’s automatic cookie submission for cross-origin requests to issue state changing requests on the user’s behalf.
```
Comment (willeponken) [inline doc/architecture/security.md]: `manin` -> `man in`
Or maybe `man-in-the-middle`?

File: doc/architecture/security.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+In order to increase the security against CSRSF attacks we have chosen to use a rolling random token scheme. This means that the base senario for protection against CSRF attacks gives you a session wide CSRF-cookie and which generates a session wide CSRF-token.
+The enhanced implementation still gives you a session wide CSRF-cookie which is used to generate _per request_ CSRF-tokens.
+
+Secure cookies
+--------------
+Roast.software cookies is stored in the cookie store in the browser. These are validate with the HMAC mechanism and encrypted with AES256.
+
+Password storage
+----------------
+There's alot of challenges keeping user credentials secured. Our solution is based on well known algorithms, Bcrypt and SHA512. The implementation is made of the official libraries for go, so that we don't make the mistake of "rolling our own crypto". The methodology used is to first hash the password with SHA512, and then strongly hashing it with Bcrypt.
```
Comment (willeponken) [inline doc/architecture/security.md]: > It's implemented using the official libraries in Go?

File: doc/architecture/security.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Below quote is found in `/model/user.go`.
```
Comment (willeponken) [inline doc/architecture/security.md]: > The quote below [...] ?

File: doc/architecture/security.md (hunk 1 above)
Comment (Hjortsberg) [reply doc/architecture/security.md]: Don't you like the sound of manin? It gives a certain feeling to it i would say. 😃
Will fix.

**Pull request merged**